### DISCUSSION TITLE
interaction with native API's

### DISCUSSION THREAD

**Discussion by bitmvk** [flet-dev/flet#3181] [Ideas]
**Please Describe The Problem To Be Solved**

the ability to use the native API's provided by the platform.

**Comment by FeodorFitsner** [comment]
What APIs, in the order of priority, are you thinking of?

**Reply by bitmvk** [reply]
in no particular order:

- Battery
- Call
- Camera 
- Flash
- Notification
- Screenshot
- GPS
- IR Blaster
- SMS (sending)
- Speech to text
- Text to speech
- Passcode(for private information) -> Face Id, Fingerprint, Passcode

are all the API's i think are required by a full-fledged UI kit.

but the main things i would like to be included are:
 - camera
 - GPS
 - Flash
 - Notification
 - Screenshot
 - Speech to text
 - Text to speech
 - Call
 - Passcode

**Reply by FeodorFitsner** [reply]
Great!

* Camera is in progress: https://github.com/flet-dev/flet/issues/1281, https://github.com/flet-dev/flet/tree/camera
* GPS: Does that count: https://github.com/flet-dev/flet/pull/3179 ?
* Flash - [torch_light](https://pub.dev/packages/torch_light) or [torch_controller](https://pub.dev/packages/torch_controller) - need an issue.
* Notification - need to investigate. Do I correctly understand that some sort of backend is required, such as Firebase, for push notifications to work?
* Screenshot - [native_screenshot](https://pub.dev/packages/native_screenshot) - need an issue.
* Speech to text - [speech_to_text](https://pub.dev/packages/speech_to_text) - need an issue.
* Text to speech - [flutter_tts](https://pub.dev/packages/flutter_tts) - need an issue.
* Call - already supported with `page.launch_url()`: https://stackoverflow.com/a/45523502/1435891 ?
* Passcode - could you please clarify?
* SMS - [flutter_sms](https://pub.dev/packages/flutter_sms) - need an issue.
* IR sensor: [ir_sensor_plugin](https://pub.dev/packages/ir_sensor_plugin) - this?

**Reply by bitmvk** [reply]
Thanks for the quick reply,

 with Camera out of the way,
I saw the GPS implementation you mentioned, and it was great!!
I thought implementing Calling would need some new control but apparently not. `page.launch_url` will get the work done.

There are 2 types of Notifications, 
 - Local Notifications([flutter_local_notifications](https://pub.dev/packages/flutter_local_notifications#-supported-platforms) or [awsome_notifications](https://pub.dev/packages/awesome_notifications#-how-to-show-local-notifications) or https://developer.android.com/develop/ui/views/notifications) 

 -  Push Notifications([firebase_messaging](https://pub.dev/packages/firebase_messaging) or [Firebase Cloud Messaging](https://firebase.flutter.dev/docs/messaging/overview)).

 Local Notifications Don't require any backend but Push Notifications do. 
The usual for Push Notification's backend is Firebase Cloud Messaging. 
I personally would prefer both to be present but at least if we have Local Notifications, it would be good enough(the tradeoff being that the app wont be able to send notification when closed. But i think there is a workaround by using services. Needs more research)

By, Passcode i meant local authentication when delicate data is to be shown. Like in a Credit Card App. i think [local_auth](https://pub.dev/packages/local_auth) will help.

Should i open the issues(feature request) for the remaining?

**Reply by bitmvk** [reply]
And for battery i think adding a property for page will do it. Like `page.battery`

**Reply by FeodorFitsner** [reply]
Yes, please open issues for everything (please make sure there are no dups) - that will be very helpful. Thank you! 🙏🏻

**Reply by bitmvk** [reply]
opened issues for all the requested features
